File: CPP/Program/Komputer.cpp
#pragma once
#include <iostream>
#include <vector>
#include <string>
#include "Cpu.cpp"
#include "Gpu.cpp"
#include "PSU.cpp"
#include "Ram.cpp"
#include "Motherboard.cpp"
#include "Penyimpanan.cpp"
#include "SistemOperasi.cpp"
#include "Pendingin.cpp"
#include "Keyboard.cpp"
#include "Mouse.cpp"
#include "Monitor.cpp"
#include "Speaker.cpp"

using namespace std;

class Komputer
{
private:
    string nama;
    Cpu cpu;
    Gpu gpu;
    Psu psu;
    Ram ram;
    Motherboard motherboard;
    Penyimpanan penyimpanan;
    SistemOperasi os;
    Pendingin pendingin;
    vector<Periferal *> periferal;

public:
    Komputer()
    {
    }

    Komputer(Cpu cpu, Gpu gpu, Psu psu, Ram ram, Motherboard motherboard, Penyimpanan penyimpanan, SistemOperasi os, Pendingin pendingin, vector<Periferal *> periferalList)
    {
        this->cpu = cpu;
        this->gpu = gpu;
        this->psu = psu;
        this->ram = ram;
        this->motherboard = motherboard;
        this->penyimpanan = penyimpanan;
        this->os = os;
        this->pendingin = pendingin;
        this->periferal = periferalList;
    }

    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setPeriferal(Periferal *periferal)
    {
        this->periferal.push_back(periferal);
    }

    string getNama()
    {
        return this->nama;
    }

    Cpu getCpu()
    {
        return this->cpu;
    }

    Gpu getGpu()
    {
        return this->gpu;
    }

    Psu getPsu()
    {
        return this->psu;
    }

    Ram getRam()
    {
        return this->ram;
    }

    Motherboard getMotherboard()
    {
        return this->motherboard;
    }

    Penyimpanan getPenyimpanan()
    {
        return this->penyimpanan;
    }

    SistemOperasi getOs()
    {
        return this->os;
    }

    Pendingin getPendingin()
    {
        return this->pendingin;
    }

    vector<Periferal *> getPeriferal()
    {
        return this->periferal;
    }

    void tambahPeriferal(Periferal *periferal)
    {
        this->periferal.push_back(periferal);
    }

    void tampilkanSpesifikasi()
    {
        cout << "Nama Komputer: " << this->nama << endl;
        cout << "Spesifikasi Komputer: " << endl;
        cout << "CPU: " << this->cpu.getNama() << " (" << this->cpu.getJumlahCore() << " Core, " << this->cpu.getKecepatanGHz() << " GHz)" << endl;
        cout << "GPU: " << this->gpu.getNama() << " (" << this->gpu.getVramGB() << " GB VRAM)" << endl;
        cout << "PSU: " << this->psu.getNama() << " (" << this->psu.getDayaWatt() << " Watt, " << this->psu.getSertifikasi() << ")" << endl;
        cout << "RAM: " << this->ram.getNama() << " (" << this->ram.getKapasitasGB() << " GB, " << this->ram.getDdr() << ")" << endl;
        cout << "Motherboard: " << this->motherboard.getNama() << " (" << this->motherboard.getSocket() << ", " << this->motherboard.getChipset() << ", " << this->motherboard.getJumlahSlotRam() << " Slot RAM)" << endl;
        cout << "Penyimpanan: " << this->penyimpanan.getNama() << " (" << this->penyimpanan.getKapasitasGB() << " GB, " << this->penyimpanan.getTipe() << ")" << endl;
        cout << "Sistem Operasi: " << this->os.getNama() << " (" << this->os.getVersi() << ", " << this->os.getBerlisensi() << ")" << endl;
        cout << "Pendingin: " << this->pendingin.getNama() << " (" << this->pendingin.getTipe() << ", " << this->pendingin.getKecepatanRPM() << " RPM)" << endl;
        cout << "Periferal: " << endl;
        for (int i = 0; i < this->periferal.size(); i++)
        {
            cout << " - " << this->periferal[i]->getNama() << " (" << this->periferal[i]->getMerk() << ", " << this->periferal[i]->getKoneksi() << ")" << endl;
        }
    }

    ~Komputer()
    {
    }
};

File: Java/Program/Komputer.java
import java.util.List;


public class Komputer 
{
    private String nama;
    private Cpu cpu;
    private Gpu gpu;
    private Ram ram;
    private PSU psu;
    private Motherboard motherboard;
    private Penyimpanan penyimpanan;
    private SistemOperasi sistemOperasi;
    private Pendingin pendingin;
    private List<Periferal> periferalList;

    public Komputer(String nama, Cpu cpu, Gpu gpu, Ram ram, PSU psu, Motherboard motherboard, Penyimpanan penyimpanan, SistemOperasi sistemOperasi, Pendingin pendingin, List<Periferal> periferalList) {
        this.nama = nama;
        this.cpu = cpu;
        this.gpu = gpu;
        this.ram = ram;
        this.psu = psu;
        this.motherboard = motherboard;
        this.penyimpanan = penyimpanan;
        this.sistemOperasi = sistemOperasi;
        this.pendingin = pendingin;
        this.periferalList = periferalList;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getNama() {
        return nama;
    }

    public void setCpu(Cpu cpu) {
        this.cpu = cpu;
    }

    public Cpu getCpu() {
        return cpu;
    }

    public void setGpu(Gpu gpu) {
        this.gpu = gpu;
    }

    public Gpu getGpu() {
        return gpu;
    }

    public void setRam(Ram ram) {
        this.ram = ram;
    }

    public Ram getRam() {
        return ram;
    }

    public void setPsu(PSU psu) {
        this.psu = psu;
    }

    public PSU getPsu() {
        return psu;
    }

    public void setMotherboard(Motherboard motherboard) {
        this.motherboard = motherboard;
    }

    public Motherboard getMotherboard() {
        return motherboard;
    }

    public void setPenyimpanan(Penyimpanan penyimpanan) {
        this.penyimpanan = penyimpanan;
    }

    public Penyimpanan getPenyimpanan() {
        return penyimpanan;
    }

    public void setSistemOperasi(SistemOperasi sistemOperasi) {
        this.sistemOperasi = sistemOperasi;
    }

    public SistemOperasi getSistemOperasi() {
        return sistemOperasi;
    }

    public void setPendingin(Pendingin pendingin) {
        this.pendingin = pendingin;
    }

    public Pendingin getPendingin() {
        return pendingin;
    }

    public void setPeriferalList(List<Periferal> periferalList) {
        this.periferalList = periferalList;
    }

    public List<Periferal> getPeriferalList() {
        return periferalList;
    }

    public void tambahPeriferal(Periferal periferal) {
        periferalList.add(periferal);
    }

    public void hapusPeriferal(Periferal periferal) {
        periferalList.remove(periferal);
    }

    public void tampilkanSpesifikasi() {
        System.out.println("Nama Komputer: " + nama);
        System.out.println("Spesifikasi Komputer:");
        System.out.println("CPU: " + cpu.getNama() + " (" + cpu.getJumlahCore() + " Core, " + cpu.getKecepatanGHz() + " GHz)");
        System.out.println("GPU: " + gpu.getNama() + " (" + gpu.getVramGB() + " GB VRAM)");
        System.out.println("PSU: " + psu.getNama() + " (" + psu.getDayaWatt() + " Watt, " + psu.getSertifikasi() + ")");
        System.out.println("RAM: " + ram.getNama() + " (" + ram.getKapasitasGB() + " GB, " + ram.getDdr() + ")");
        System.out.println("Motherboard: " + motherboard.getNama() + " (" + motherboard.getSocket() + ", " + motherboard.getChipset() + ", " + motherboard.getJumlahSlotRam() + " Slot RAM)");
        System.out.println("Penyimpanan: " + penyimpanan.getNama() + " (" + penyimpanan.getKapasitasGB() + " GB, " + penyimpanan.getTipe() + ")");
        System.out.println("Sistem Operasi: " + sistemOperasi.getNama() + " (" + sistemOperasi.getVersi() + ", " + sistemOperasi.getBerlisensi() + ")");
        System.out.println("Pendingin: " + pendingin.getNama() + " (" + pendingin.getMerk() + ", " + pendingin.getTipe() + ")");
        
        System.out.println("Periferal:");
        for (Periferal periferal : periferalList) {
            System.out.println(" - " + periferal.getNama() + " (" + periferal.getMerk() + ", " + periferal.getKoneksi() + ")");
        }
    }

}